Core/clim-core/gadgets.lisp

```diff
@@ -1659,13 +1661,22 @@ and must never be nil.")
 
 (defmethod compose-space ((pane slider-pane) &key width height)
   (declare (ignore width height))
-  (let ((minor (+ 8 (* 2 4) (if (gadget-show-value-p pane) 30 0)))
-        (major 128))
-    (if (eq (gadget-orientation pane) :vertical)
-        (make-space-requirement :min-width  minor :width  minor
-                                :min-height major :height major)
-        (make-space-requirement :min-width  major :width  major
-                                :min-height minor :height minor))))
+  (multiple-value-bind (vw vh)
+      (text-size pane (format-value (gadget-min-value pane) (slider-decimal-places pane)))
+    (setq vw
+      (max vw
+           (text-size pane
+                      (format-value (gadget-max-value pane) (slider-decimal-places pane)))))
+    (let ((minor
+            (if (gadget-show-value-p pane)
+                (* 2 (+ 10.0 (if (eq (gadget-orientation pane) :vertical) vw vh))) ; the value
```

Comment:
Please avoid (by defining constants) or explain the new magic numbers.

Reply:
These values (2, 10.0, 8.0 and 1+) tie in with the slider-pane handle-repaint method, so yes, it would improve readability to make some of them constants and change handle-repaint accordingly.

Reply:
Given that #647 contains some refactoring of the slider gadget, maybe it is fine to leave the magic numbers for now.

Reply:
#647 has been merged.

Reply:
same concern is voiced against new code, resolving this conversation.